onnxruntime/core/providers/cuda/cu_inc/unary_elementwise_impl.cuh

```diff
@@ -9,17 +9,21 @@
 namespace onnxruntime {
 namespace cuda {
 
-template <typename InT, typename OutT, typename FuncT>
+template <typename InT, typename OutT, typename FuncT, int NumThreadsPerBlock, int NumElementsPerThread>
 __global__ void _UnaryElementWise(
     const InT* input_data,
     OutT* output_data,
     const FuncT& functor,
     CUDA_LONG N) {
-  if (N == 0)  // special case where there's a dim value of 0 in the shape
-    return;
+  CUDA_LONG id = NumElementsPerThread * blockDim.x * blockIdx.x + threadIdx.x;
 
```

Comment:
This was only just added to handle downstream nodes from a NonZero op where the number of matches is 0, leading to a dim value of 0 in the shape. How are you handling that if you're removing this?

Reply:
We don't need this check in the GPU kernel, as we already check it before we launch the kernel.